content/articles/2018-11-tibble-2.0.0-pre-announce.Rmd

```
@@ -0,0 +1,222 @@
+
+## Deprecation of `validate` in `as_tibble()`
+
+In tibble v1.4.2, `as_tibble()` has a `validate` argument, but its default behaviour value was inconsistent across different methods and there was no equivalent argument for `tibble()`. The `validate` argument is now soft-deprecated and its use will trigger a message, once per session. The `validate` argument will eventually be removed, but for now it can be used jointly with the new `.name_repair` argument (without even triggering a message). This is possible, because fortunately the `.name_repair` argument to `as_tibble()` is ignored in tibble v1.4.2.
+
+Here's what `validate` does in the development version of tibble for tibbles, data frames, and matrices, along with suggested alternatives.
+
+### Tibbles and data frames
+
+The default was `validate = FALSE` for tibbles and `validate = TRUE` for data frames.
+Code that worked before for tibbles can now throw unexpected errors if the resulting tibble has problematic names.
+Use the default instead of `validate = TRUE`, and `.name_repair = "minimal"` in addition to `validate = FALSE`.
+If your code targets tibble >= v2.0.0, you can remove the `validate` argument.
+
+```{r}
+df <- new_tibble(list(a = 5, a = 6), nrow = 1)
+
+# errors, as it should, because names are duplicated ... but also messages
+as_tibble(df, validate = TRUE)
+
+# errors due to default .name_repair = "check_unique"
+# (but no error in tibble v1.4.2)
+as_tibble(df)
+
+# ensures that the validate = TRUE default is used for tibble < 2.0.0
+as_tibble(as.data.frame(df))
+
+
+# no error ... but still messages
+as_tibble(df, validate = FALSE)
+
+# no error, quietly
+as_tibble(df, .name_repair = "minimal")
+
+# no error, quietly, compatible with tibble < 2.0.0
+as_tibble(df, validate = FALSE, .name_repair = "minimal")
+```
+
+
+### Matrices and other objects
+
+The `validate` argument was silently ignored in v1.4.2, now triggers a message.
```

maybe offset the reference to 1.4.2 ⇨ something like:
"The `validate` argument, which was silently ignored in v1.4.2, now triggers a message."